eng/scripts/docs/Docs-ToC.ps1

```diff
@@ -21,3 +21,188 @@ function Get-NamepspacesFromDll($dllPath) {
 }
+
+function DownloadNugetPackage($package, $version, $destination) {
+    # $PackageSourceOverride is a global variable provided in
+    # Update-DocsMsToc.ps1. Its value can set a "customSource" property.
+    # If it is empty then the property is not overridden.
+    $customPackageSource = Get-Variable -Name 'PackageSourceOverride' -ValueOnly -ErrorAction 'Ignore'
+    # Download package from nuget, if no package found in nuget, then search devops public feeds if set.
+    Write-Host "Downloading $package with version $version..."
+    if ($customPackageSource) {
+        nuget install -FallbackSource $customPackageSource $package -Version $version -DirectDownload -DependencyVersion Ignore -OutputDirectory $destination
+    }
+    else {
+        nuget install $package -Version $version -DirectDownload -DependencyVersion Ignore -OutputDirectory $destination
+    }
+}
+
+function Fetch-NamespacesFromNupkg ($package, $version) {
+    $tempLocation = (Join-Path ([System.IO.Path]::GetTempPath()) "extractNupkg")
+    if (!(Test-Path $tempLocation)) {
+        New-Item -ItemType Directory -Path $tempLocation -Force | Out-Null
+    }
+    Write-Host "Downloading nupkg package $package with version $version to $tempLocation ...."
+    DownloadNugetPackage -package $package -version $version -destination $tempLocation | Out-Null
+    $packageFolder = "$package.$version"
+    if ($LASTEXITCODE -ne 0 -or !$(Test-Path $tempLocation/$packageFolder)) {
+        Write-Host "Did not download the package correctly. Skipping..."
+        return @()
+    }
+    Write-Host "Searching for dll file..."
+    $dllFile = @()
+    if (Test-Path "$tempLocation/$packageFolder/lib/netstandard2.0/"){
+        $dllFile = @(Get-ChildItem "$tempLocation/$packageFolder/lib/netstandard2.0/*" -Filter '*.dll' -Recurse)
+    }
+    if (!$dllFile) {
+        Write-Warning "Can't find any dll file from $tempLocation/$packageFolder with target netstandard2.0."
+        $dllFiles = Get-ChildItem "$tempLocation/$packageFolder/lib/*" -Filter '*.dll' -Recurse
+        if (!$dllFiles) {
+            Write-Error "Can't find any dll file from $tempLocation/$packageFolder." -ErrorAction Continue
+            return @()
+        }
+        $dllFile = $dllFiles[0]
+    }
+    elseif ($dllFile.Count -gt 1) {
+        Write-Warning "There are multiple dll files in target netstandard2.0. Choose the one with package name."
+        if (Test-Path "$tempLocation/$packageFolder/lib/netstandard2.0/$package.dll") {
+            Write-Host "Use the dll file $package.dll"
+            $dllFile = [System.IO.DirectoryInfo]"$tempLocation/$packageFolder/lib/netstandard2.0/$package.dll"
+        }
+        else {
+            $dllFile = $dllFile[0]
+        }
+    }
+    Write-Host "Dll file found: $($dllFile.FullName)"
+    $namespaces = Get-NamespacesFromDll $dllFile.FullName
+    if (!$namespaces) {
+        Write-Error "Can't find namespaces from dll file $($dllFile.FullName)." -ErrorAction Continue
```

Comment:
Why are you setting ErrorAction Continue on these Write-Error messages? Also should we should probably be using the LogError helpers in these cases so they output correctly as errors in DevOps.

Reply:
I have one testing failed in the middle. I switched to logging.ps1 apis if it's non stop behavior.

Reply:
Tested in local. The tests failed at Log function in logging.ps1 since we did not import the logging script in test script. I switched back the Write-Error